.github/workflows/bump-version.yml

```diff
@@ -0,0 +1,132 @@
+name: Bump version
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      newVersion:
+        description: Version to bump
+        required: true
+        default: auto
+        type: string
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  VERSION_FILE: ./package.json
+
+jobs:
+  set-next-version:
+    name: Set next version automatically
+    runs-on: ubuntu-latest
+    if: github.event_name != 'workflow_dispatch' || github.event.inputs.newVersion == 'auto'
+    outputs:
+      version: ${{ steps.set-next-version.outputs.version }}
+
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Get version on main
+        id: get-main-version
+        run: |
+          currentVersion=$(node -p -e "require('${{ env.VERSION_FILE }}').version")
+          echo "version=$currentVersion" >> $GITHUB_OUTPUT
+
+      - name: Checkout develop
+        uses: actions/checkout@v4
+        with:
+          ref: develop
+
+      - name: Get version on develop
+        id: get-develop-version
+        run: |
+          currentVersion=$(node -p -e "require('${{ env.VERSION_FILE }}').version")
+          echo "version=$currentVersion" >> $GITHUB_OUTPUT
+
+      - name: Set next version
+        id: set-next-version
+        run: |
+          function ver { printf "%03d%03d%03d%03d" $(echo "$1" | tr '.' ' '); }
+
+          echo "main version: $(ver ${{ steps.get-main-version.outputs.version }})"
+          echo "develop version: $(ver ${{ steps.get-develop-version.outputs.version }})"
+
+          if [[ $(ver ${{ steps.get-main-version.outputs.version }}) -gt $(ver ${{ steps.get-develop-version.outputs.version }}) ]]; then
+            echo "main version is greater than develop version"
+
+            echo "version=${{ steps.get-main-version.outputs.version }}" >> $GITHUB_OUTPUT
+          else
+            echo "develop version is greater or equal to main version"
+
+            delimiter=.
```

Comment:
This deleimiter can be considered as an en var. Also deifning is as such would make it easier to see as I almost did not see that dot on a first pass:

```
VERSION_DELIMITER: "."
```

Reply:
Refactored in https://github.com/nimblehq/infrastructure-templates/pull/237/commits/790a7d8f4c39523afe36a311082e94200773b2f7